compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2045,7 +2046,7 @@ object Parsers {
     def expr1Rest(t: Tree, location: Location): Tree = in.token match
       case EQUALS =>
         t match
-          case Ident(_) | Select(_, _) | Apply(_, _) =>
+          case Ident(_) | Select(_, _) | Apply(_, _) | PrefixOp(_, _) =>
```

Comment:
What must have happened in Scala 2: We had similar logic here, but we did not mark prefix operations specially; so
a prefix operation was accepted since it was represented by a `Select`. Similarly for infix. The following expression
is parsed in Scala 2, but will get a type error afterwards:
```scala
   1 + 1 = 2
```
In dotc you get instead a parse error:
```
-- Error: ../../new/test.scala:4:10 --------------------------------------------
4 |    1 + 1 = 2
  |          ^
  |          end of statement expected but '=' found
```
 That's because we represent infix operations as `Infix` nodes instead of `Apply` nodes.